rust/main/lander/src/adapter/chains/radix/adapter.rs

```diff
@@ -71,16 +227,132 @@ impl AdaptsChain for RadixAdapter {
         build_txs
     }
 
-    async fn simulate_tx(&self, _tx: &mut Transaction) -> Result<Vec<PayloadDetails>, LanderError> {
-        todo!()
+    async fn simulate_tx(&self, tx: &mut Transaction) -> Result<Vec<PayloadDetails>, LanderError> {
+        tracing::info!(?tx, "simulating transaction");
+
+        let (visible_components, fee_summary) = {
+            let tx_precursor = tx.precursor();
+            // decode manifest value from Mailbox::process_calldata()
+            let manifest_value: ManifestValue = manifest_decode(&tx_precursor.encoded_arguments)
+                .map_err(|err| {
+                    let error_msg = "Failed to decode manifest";
+                    tracing::error!(?err, "{error_msg}");
+                    LanderError::PayloadNotFound
+                })?;
+
+            let manifest_args = match manifest_value {
+                sbor::Value::Tuple { fields } => fields,
+                _ => vec![],
+            };
+
+            let decoder = AddressBech32Decoder::new(&self.network);
+            let component_address =
+                ComponentAddress::try_from_bech32(&decoder, &tx_precursor.component_address)
+                    .ok_or_else(|| {
+                        let error_msg = "Failed to parse ComponentAddress";
+                        tracing::error!(
+                            component_address = tx_precursor.component_address,
+                            "{error_msg}"
+                        );
+                        LanderError::PayloadNotFound
+                    })?;
+            self.visible_components(&component_address, &tx_precursor.method_name, manifest_args)
+                .await?
+        };
+
+        let precursor = tx.precursor_mut();
+
+        precursor.fee_summary = Some(fee_summary);
+        precursor.visible_components = Some(VisibleComponents {
+            addresses: visible_components.iter().map(|v| v.to_hex()).collect(),
+        });
+        Ok(Vec::new())
     }
 
-    async fn estimate_tx(&self, _tx: &mut Transaction) -> Result<(), LanderError> {
-        todo!()
+    async fn estimate_tx(&self, tx: &mut Transaction) -> Result<(), LanderError> {
+        self.simulate_tx(tx).await?;
+        Ok(())
```

Comment:
Would it make sense to implement method `estimate_gas_limit` as well? Possibly, in the future PR?